src/UIHandler.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#ifndef UIHANDLER_H
+#define UIHANDLER_H
+
+#include "debug.h"
+
+#include "cursesdef.h"
+#include "output.h"
+
+struct IntPair
+{
+    int x;
+    int y;
+};
+
+class UIPanel
+{
+public:
+    virtual std::vector<UIPanel*> getChild() = 0;
+    virtual void addChild(UIPanel *panel) = 0;
```

Comment:
Here for example it's not clear who owns the `panel` pointer and who is responsible for deleting it.

You could write this as `void addChild( std::unique_ptr<UIPanel> panel );`, which clearly states the function will take care of the pointer (either by storing it or by deleting it).